### DISCUSSION TITLE
How to update the Workflow Permissions?

### DISCUSSION THREAD

**Discussion by gargajit** [fastn-stack/fastn#596] [Q&A] [answered]
When using [fastn-template](https://github.com/ftd-lang/fastn-template), why is the Initial Commit workflow failing after creating a repository?

**Comment by gargajit** [comment]
This is because Workflow Permissions by default is Read-only. 
For Initial Commits or if you update anything in the repository, you need Read and Write Permissions on GitHub.

To change the Workflow Permissions, please follow the following steps:
1. In your package on GitHub, go to Settings
2. In the Table of Content, on the left side, click on the Actions drop-down and select General
3. Scroll down to Workflow Permissions and select Read and write Permissions and Save it.

Now you can go to Actions and re-run your failed workflows.

**Comment by amitu** [comment, marked as answer]
The [permission issue in workflow is fixed](https://github.com/ftd-lang/fastn-template/commit/ecb567c88327c667ca7400d800dc86509fe120ad), it should not happen again, no need to grant permission via settings any more.
